### ISSUE TITLE
Update INSTALL.md?

### ISSUE THREAD

**Issue by mikeshulman** [HoTT/HoTT#929]
Currently `INSTALL.md` instructs Linux users to use their package manager to install ocaml, camlp5, etc.  Is this still the preferred approach?  I had issues with the recent bump of Coq until I blew away all my apt packages and installed everything with opam instead.

**Comment by spitters** [comment]
I'm having trouble upgrading too.

etc/install_coq.sh
....
You have OCaml 4.06.0. Good!
You have Camlp5 7.05. Good!
....

File "tools/coqdep_lexer.mll", line 174, characters 22-41:
Warning 3: deprecated: String.uncapitalize
Use String.uncapitalize_ascii instead.
File "tools/coqdep_lexer.mll", line 356, characters 18-37:
Warning 3: deprecated: String.uncapitalize
Use String.uncapitalize_ascii instead.
File "tools/coqdep_lexer.mll", line 361, characters 28-47:
Warning 3: deprecated: String.uncapitalize


Not sure what is going on.

**Comment by mikeshulman** [comment]
Those just look like warnings not errors?

**Comment by spitters** [comment]
I'm making a mess. I should have used
```
opam install coq-hott
```

Will come back when it works.

**Comment by spitters** [comment]
Yes, that works without problems.
I have the ocaml tools installed via opam.
We should indeed update INSTALL.ms

**Comment by mikeshulman** [comment]
Okay, well someone should update INSTALL.md.  I'm not a good person to do it because I don't know much about how opam works.

**Comment by spitters** [comment]
#930

**Comment by u5943321** [comment]
I am an OS X user and the INSTALL.md does not seem to work as well. In particular, my problem is https://groups.google.com/forum/#!topic/hott-cafe/68XmcNGtMIY Could someone please help? I am trying to launching into using the Coq HoTT library, and I cannot install the IDE. As in the post on HoTT cafe, Scott has the same problem as me.

**Comment by mikeshulman** [comment]
There must be some OS X users around somewhere...

**Comment by mseri** [comment]
@u5943321 Try installing libffi and gtksourceview (via homebrew and macports) and then run
```
PKG_CONFIG_PATH='path/to/libffi/pkgconfg' opam install ...
```
You may also need to append `:path/to/gktsourceview/pkgconfig` or more to the path, but it seems to work with just libffi for me

**Comment by McTano** [comment]
I was at the HoTT Summer School at CMU earlier this month, and was able to help a number of people using Ubuntu (and Mint) get the installation to work. I put together some additional instructions for linux users, which might help with updating the `install_coq_deps.sh` script and the linux instructions in `INSTALL.md`.

@mikeshulman I see you've done some work on this in your un-merged PR #1025 from April.

I agree that it's probably best to merge the instructions for MacOS and Linux, especially the parts dealing with OPAM, which are currently missing from the linux instructions.

I'll post my notes below, which include solutions for the issues identified in #1040 and #1021.

# Additional instruction notes for Linux Users

## System dependecies
- The gtk-related dependencies in the script `etc/install_coq_deps.sh` are out of date. You need the gtk3 versions instead of the gtk2 ones.
- Here are alternate commands to get the system dependencies for Ubuntu users.
```
sudo apt-get update
sudo apt-get install curl sed grep wget tar libgtk-3-dev libgtk3sourceview-3.0-dev libexpat1-dev 
```

## opam dependencies
- Run `opam --version` to check that opam is installed and up to date. If the opam version is < 2, you should install opam, using the curl command provided at https://opam.ocaml.org/.
```
    sh <(curl -sL https://raw.githubusercontent.com/ocaml/opam/master/shell/install.sh)
```

- You should then run `opam init` to make sure opam is initialized. I recommend allowing opam to change your `.profile` file when it asks for permission.
- To install the required ocaml libraries, run 
```
opam install ocaml camlp5 lablgtk3 lablgtk3-sourceview3 ocamlfind num
```

## The rest
- The steps above should be performed before running `/etc/install.sh`
- After this, you should be able to follow the rest of the steps in order.

## Bonus Tips
- When you execute `etc/install.sh`, you can look at the output to see the result of a series of checks it performs at the beginning of the installation.
- If opam complains that you do not have a switch installed, run ```opam switch create 4.0.7```
